src/main/java/com/hoteldetails/pad/client/RoomAvailabilityClient.java

```diff
@@ -0,0 +1,50 @@
+package com.hoteldetails.pad.client;
+
+
+import com.hoteldetails.pad.endpoint.RoomAvailabilityDjocaEndpointFactory;
+import lombok.extern.slf4j.Slf4j;
+import org.opentravel.ota._2003._05.PropertyAvailabilityRQ;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.bind.Unmarshaller;
+import java.io.StringReader;
+import java.io.StringWriter;
+
+@Component
+@Slf4j
+public class RoomAvailabilityClient {
+
+    public Object restClient(PropertyAvailabilityRQ propertyAvailabilityRQ, String supplierUrl, String service) throws JAXBException {
+
+        JAXBContext context= RoomAvailabilityDjocaEndpointFactory.context;
+        final StringWriter requestWriter = new StringWriter();
+        Marshaller marshaller = context.createMarshaller();
+        Unmarshaller unmarshaller = context.createUnmarshaller();
+        try {
+            marshaller.marshal(propertyAvailabilityRQ, requestWriter);
+            RestTemplate restTemplate = new RestTemplate();
+            UriComponentsBuilder uriBuilder = UriComponentsBuilder.fromHttpUrl(supplierUrl);
+            uriBuilder.path(service);
+            ResponseEntity<String> responseEntity = restTemplate.postForEntity(uriBuilder.build().toUriString(),requestWriter.toString(), String.class);
+            log.info(requestWriter.toString());
+            log.info(responseEntity.getBody());
+            return unmarshaller.unmarshal(new StringReader(responseEntity.getBody()));
+        }
+        catch (JAXBException b){
+            log.info("JAXBException caught" +b);
+            throw b;
+        }
+        catch (Exception e)
+        {
+            log.info("Exception occured in request-response to Djoca" +e);
+            throw e;
+        }
+    }
+
+}
```

Comment:
No need to create a seperate client for room availability . You can re use DjocaClient.java